### PR TITLE
Feature/clang tidy

### PR DESCRIPTION
**Pull request by JonasGilg**

### CODE REVIEW

File: src/LODVisitor.cpp [hunk 1]
```diff
@@ -63,8 +63,8 @@ bool testInFrustum(Frustum const& frustum, BoundingBox<double> const& tb) {
 
     // test if any BB corner is inside the halfspace defined
     // by the current plane
-    for (std::size_t j = 0; j < tbPnts.size(); ++j) {
-      if (glm::dot(normal, tbPnts[j]) >= d) {
+    for (auto& tbPnt : tbPnts) {
```
Comment (Schneegans) [inline src/LODVisitor.cpp]: Can we use const& here?

```suggestion
    for (auto const& tbPnt : tbPnts) {
```

File: src/LODVisitor.cpp [hunk 2]
```diff
@@ -113,24 +114,29 @@ bool testFrontFacing(glm::dvec3 const& camPos, PlanetParameters const* params,
           glm::dvec3(tbMax[0], tbMax[1], tbMax[2]), glm::dvec3(tbMin[0], tbMax[1], tbMax[2])}};
 
   // Simple ray-sphere intersection test for every corner point
-  for (std::size_t j = 0; j < tbPnts.size(); ++j) {
-    double     dRayLength = glm::length(tbPnts[j] - camPos);
-    glm::dvec3 vRayDir    = (tbPnts[j] - camPos) / dRayLength;
+  for (auto& tbPnt : tbPnts) {
```
Comment (Schneegans) [inline src/LODVisitor.cpp]: Can we use const& here?

```suggestion
  for (auto const& tbPnt : tbPnts) {
```

File: src/PlanetShader.cpp [hunk 3]
```diff
@@ -34,52 +36,51 @@ std::map<std::string, cs::graphics::ColorMap> PlanetShader::mColorMaps;
...
+        [this](bool /*ignored*/) { mShaderDirty = true; });
   // clang-format on
 
   // TODO: color map mangement could be done in a separate class
   if (mColorMaps.empty()) {
     auto files(cs::utils::filesystem::listFiles("../share/resources/colormaps"));
 
     bool first = true;
-    for (auto file : files) {
+    for (const auto& file : files) {
```
Comment (Schneegans) [inline src/PlanetShader.cpp]:
```suggestion
    for (auto const& file : files) {
```

File: src/Plugin.cpp [hunk 4]
```diff
@@ -20,6 +20,18 @@
...
+
+#ifdef min
+#undef min
+#endif
+
+#ifdef max
+#undef max
+#endif
+
+#endif
```
Comment (Schneegans) [inline src/Plugin.cpp]: Isn't that handled by our `-DNOMINMAX`?

File: src/RenderDataDEM.cpp [hunk 5]
```diff
@@ -13,11 +13,11 @@
 namespace csp::lodbodies {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
-
-/* static */ RenderDataDEM* RenderDataDEM::create(TileNode* node) {
+/*
+RenderDataDEM* RenderDataDEM::create(TileNode* node) {
```
Comment (Schneegans) [inline src/RenderDataDEM.cpp]: Can this be removed?

File: src/RenderDataImg.cpp [hunk 6]
```diff
@@ -9,11 +9,11 @@
 namespace csp::lodbodies {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
-
-/* static */ RenderDataImg* RenderDataImg::create(TileNode* node) {
+/*
+RenderDataImg* RenderDataImg::create(TileNode* node) {
```
Comment (Schneegans) [inline src/RenderDataImg.cpp]: Remove?

File: src/TileBounds.cpp [hunk 7]
```diff
@@ -49,10 +49,10 @@ BoundingBox<double> calcTileBounds(double tmin, double tmax, int tileLevel, glm:
   // tile corners
   auto corners = hp.getCornersLngLat(patchIdx);
 
-  for (std::size_t i = 0; i < corners.size(); ++i) {
+  for (auto& corner : corners) {
```
Comment (Schneegans) [inline src/TileBounds.cpp]: Can we use const&?

```suggestion
  for (auto const& corner : corners) {
```

File: src/TileBounds.cpp [hunk 8]
```diff
@@ -63,10 +63,10 @@ BoundingBox<double> calcTileBounds(double tmin, double tmax, int tileLevel, glm:
   // tile edge center points
   auto edges = hp.getEdgeCentersLngLat(patchIdx);
 
-  for (std::size_t i = 0; i < edges.size(); ++i) {
+  for (auto& edge : edges) {
```
Comment (Schneegans) [inline src/TileBounds.cpp]: Can we use const&?
```suggestion
  for (auto const& edge : edges) {
```

File: src/TileSourceWebMapService.cpp [hunk 9]
```diff
@@ -27,6 +27,10 @@
 
 #include <tiffio.h>
 
+#if defined(_WIN32) && defined(min)
+#undef min
+#endif
```
Comment (Schneegans) [inline src/TileSourceWebMapService.cpp]: NOMINMAX?

**Review by Schneegans — CHANGES_REQUESTED** [review]

File: src/Plugin.cpp (hunk 4 above)
Comment (JonasGilg) [reply src/Plugin.cpp]: I remember clang-tidy threw an error there. It seems to be fixed now.

**Review by Schneegans — APPROVED** [review]

**Pull request merged**